engine/src/main/java/org/hibernate/validator/internal/metadata/descriptor/ConstraintDescriptorImpl.java

```diff
@@ -474,9 +479,31 @@ else if ( hasParameters && !hasReturnValue ) {
 			}
 		}
 
-		// Now we are out of luck
 		if ( constraintType == null ) {
-			throw LOG.getImplicitConstraintTargetInAmbiguousConfigurationException( annotationType );
+			// check if it's a composite constraint and is built with constraints for which a type can be detected
+			Set<ConstraintType> constraintTypes = findComposingConstraints( constraintHelper )
```

Comment:
We already call `parseComposingConstraints()`; could we somehow combine this, instead of running the annotation tree another time?

Reply:
Ok, I'll see what can be done about it and open another PR if something good comes out.